pkg/kubelet/cm/memorymanager/state/state.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	v1 "k8s.io/api/core/v1"
+)
+
+// MemoryTable contains memory information
+type MemoryTable struct {
+	TotalMemSize   uint64 `json:"total"`
+	SystemReserved uint64 `json:"systemReserved"`
+	Allocatable    uint64 `json:"allocatable"`
+	Reserved       uint64 `json:"reserved"`
+	Free           uint64 `json:"free"`
+}
+
+// NodeState contains NUMA node related information
+type NodeState struct {
+	// NumberOfAssignments contains a number memory assignments from this node
+	// When the container requires memory and hugepages it will increase number of assignments by two
+	NumberOfAssignments int `json:"numberOfAssignments"`
+	// MemoryTable contains NUMA node memory related information
+	MemoryMap map[v1.ResourceName]*MemoryTable `json:"memoryMap"`
+	// Nodes contains the current NUMA node and all other nodes that are in a group with current NUMA node
+	// This parameter indicates if the current node is used for the multiple NUMA node memory allocation
+	// For example if some container has pinning 0,1,2, NUMA nodes 0,1,2 under the state will have
+	// this parameter equals to [0, 1, 2]
+	Nodes []int `json:"nodes"`
```

Comment:
the use of nodes is confusing more i look at this.

openstack had used nodes or cells interchangeably.

https://docs.openstack.org/nova/pike/admin/cpu-topologies.html#:~:text=NUMA%20is%20a%20derivative%20of,this%20shared%20bus%20is%20limited.

i wonder if we should use cell.

Reply:
+0 (soft +1) for cell. I don't know what NUMA people call these.

Reply:
Renamed to Cells.